Kickstarter-iOS/Views/Controllers/CommentRepliesViewController.swift

```diff
@@ -98,6 +107,22 @@ final class CommentRepliesViewController: UITableViewController {
         self.tableView.reloadData()
       }
 
+    self.viewModel.outputs.loadFailableReplyIntoDataSource
+      .observeForUI()
+      .observeValues { failableComment, replaceableCommentId, project in
```

Comment:
By not using `[weak self]` here this will cause a retain cycle. I see we also missed this for `loadRepliesAndProjectIntoDataSource` so please add to both.

The purpose of the `IndexPath` being returned from the `replace` function is to allow us to only reload that row which makes for a better experience. I think we want this to look more like:

```swift
self.viewModel.outputs.loadFailableReplyIntoDataSource
  .observeForUI()
  .observeValues { [weak self] failableComment, replaceableCommentId, project in
    guard let self = self else { return }
    self.tableView.beginUpdates()

    let (indexPath, newComment) = self.dataSource.replace(
      comment: failableComment,
      and: project,
      byCommentId: replaceableCommentId
    )

    if let lastIndexPath = indexPath {
      if newComment {
        self.tableView.insertRows(at: [lastIndexPath], with: .fade)
        self.tableView.scrollToRow(at: lastIndexPath, at: .bottom, animated: true)
      } else {
        self.tableView.reloadRows(at: [lastIndexPath], with: .fade)
      }
    }

    self.tableView.endUpdates()
  }
```

Reply:
Got it! Yes I thought I was missing something here regarding insertions/reloading.

Reply:
Thanks for pointing this out!

Reply:
I used `performBatchUpdates` instead as the `UITableView` says `beginUpdates` `endUpdates` will be deprecated soon in favour of `performBatchUpdates`.  The reloading works as it should for both row insert and row reload. 

The scrolling to the bottom however needs a delay, because it cannot be done inside the `performBatchUpdates` or `beginUpdates` blocks. The error is an index out of bounds issue that states "Attempting to scroll to row 8 when only 7 rows in section". The `insertRow` is still continuing to finish when `scrollToRow` is called which causes the crash. It's not an animation issue either. Open to suggestions to avoid the delay, but it seems acceptable.